stl/inc/execution

```diff
@@ -941,6 +941,27 @@ struct _Static_partition_range<_FwdIt, _Diff, false> {
     }
 };
 
+template <class _Cancel_or_completion, class _Diff>
+_NODISCARD _Static_partition_key<_Diff> _Get_next_if_not_canceled(
+    _Cancel_or_completion& _Status, _Static_partition_team<_Diff>& _Team) {
+    // Check cancellation/completion status before getting a chunk of work to process, instead of after. If we check
+    // after and a result was found in a later block than the one we are assigned, we won't notice an earlier/better
+    // result in our block.
+
+    bool _Done;
+    if constexpr (is_same_v<_Cancel_or_completion, _Cancellation_token>) {

```

Comment:
Perhaps we should rename both of these "_Done" or something like that so that you can avoid this metaprogramming?

Reply:
How about `_Finished`, and rename the function to `_Get_next_if_not_finished`?

Reply:
Sounds fine to me. I only mentioned done because that's the language WG21-P0443 uses but that might have different semantics than we do here.

Reply:
`_Done`/`_Get_next_if_not_done` it is, then.